Borland MAKE build, WAD file asset storage

phlash opened the pull request:
Howdy, I've been playing with your delightful MSDOS/VGA game engine, and thought it might be useful to build with the Borland Make available in TurboC++, and support a packed asset file (WAD file) a-lá Doom, for easy distribution of creations.. ~~This is the first working version, it could do with a couple of extra changes to allow run-time selection of asset loading mode and WAD file path.~~

Thoughts / comments / outright revulsion - all welcome :smile:

mills32 commented:
Hi, I updated a lot of things, and also dropped EGA support, it was very slow. But I like the idea of using wad files to include level+music etc. I'll try to check your commits to see how they fit in the new version.

phlash commented:
Thanks! I tried to squeeze everything into `main.c`, and added a few pointer diversions for `stdio` functions. Probably a better idea would have been to put the WAD I/O code into it's own bit of the engine, rather than the demo game :smile: !

I added the `makefile` stuff to speed up my code-build-test-fail cycle time while working on the WAD support, should probably have been in a separate PR, but either way it's orthogonal to the WAD changes - feel free to use what works for you!

mills32 commented:
> 
> 
> Thanks! I tried to squeeze everything into `main.c`, and added a few pointer diversions for `stdio` functions. Probably a better idea would have been to put the WAD I/O code into it's own bit of the engine, rather than the demo game 😄 !
> 
> I added the `makefile` stuff to speed up my code-build-test-fail cycle time while working on the WAD support, should probably have been in a separate PR, but either way it's orthogonal to the WAD changes - feel free to use what works for you!

I want the main to be just a sample/demo, because the rest of the files are compiled into a .lib, so that in the future, you'll just compile your game (in this case, main.c) and the .lib.  The wad stuff should be in one of the other files, or its own file, like "wad.c".

mills32 commented:
> Thanks! I tried to squeeze everything into `main.c`, and added a few pointer diversions for `stdio` functions. Probably a better idea would have been to put the WAD I/O code into it's own bit of the engine, rather than the demo game 😄 !
> 
> I added the `makefile` stuff to speed up my code-build-test-fail cycle time while working on the WAD support, should probably have been in a separate PR, but either way it's orthogonal to the WAD changes - feel free to use what works for you!

I finally added a very simple container for files, I did not use WAD because it was a bit complex for this. 

My file format "DAT" is just a concatenation of files, with a little header showing the number of files and sizes inside the DAT. I created a program to generate DAT and also extract the files inside. The engine can load files normally or load files inside DAT with only a few changes in the source code.

Game folcers really look better and "retro" with their files stored in DAT :). I don't know when I will upload the changes.
